Abe/abe.py

```diff
@@ -415,11 +415,20 @@ def handle_chain(abe, page):
         body = page['body']
         body += abe.search_form(page)
 
-        count = get_int_param(page, 'count') or 20
-        if count >= 2017:
+        try:
+            count = get_int_param(page, 'count') or 20
+            if count >= 2017:
+                count = 20
+            elif count <= 0:
+                count = 20
+        except ValueError:
             count = 20
-        hi = get_int_param(page, 'hi')
-        orig_hi = hi
+
+        try:
+            hi = get_int_param(page, 'hi')
+            orig_hi = hi
```

Comment:
I should have paid more attention on my first review... that orig_hi should go after the try block - looking at the original core it was always set to the same value as hi but now won't exist after exception handling.